nexus/src/app/background/sync_service_zone_nat.rs

```diff
@@ -265,6 +284,38 @@ impl BackgroundTask for ServiceZoneNatTracker {
                 });
             }
 
+            if dns_count < MIN_EXTERNAL_DNS_COUNT {
+                error!(
+                    &log,
+                    "generated config for fewer than the minimum allowed number of dns zones";
+                );
+                return json!({
+                    "error": "generated config for fewer than the minimum allowed number of dns zones"
+                });
+            }
+
+            if ntp_count < MIN_NTP_COUNT {
+                error!(
+                    &log,
+                    "generated config for fewer than the minimum allowed number of ntp zones";
+                );
+                return json!({
+                    "error": "generated config for fewer than the minimum allowed number of ntp zones"
+
+                });
+            }
+
+            if nexus_count < MIN_NEXUS_COUNT {
```

Comment:
Example where this could go awry:

- Sled pulled (or crashes, or catches fire) we mark it as "Removed"
- We identify that the services on the sled are not present
- Inventory reports that two Nexus instances are running
- 2 < 3
- So we fail the RPW here, and never update **any** NAT entries until redundancy is fully restored?

Reply:
Yes, that is correct. Will we not attempt to move the service zone to a new sled in this situation?

Reply:
As soon as we implement service re-provisioning, this would happen, but I think the scope of "new service provisions" is going to start with only Crucible and NTP.

Until that is fully implemented, this check would just stop NAT propagation